### ISSUE TITLE
Check if dataset storage location was already registered when importing datasets

### ISSUE THREAD

**Issue by dlpzx** [awslabs/aws-dataall#321]
**Is your idea related to a problem? Please describe.**
[FORWARDED FROM CUSTOMER]
We create data pipelines and run crawlers to create tables, we use Lake Formation to grant access to ETL resources and crawlers to the tables that we are creating. The S3 location is registered in Lake Formation. 

Then we import the dataset in data.all, but the creation of the stack fails because it tries to register a data location that is already registered.

**Describe the solution you'd like**
I would like data.all to register the location only when it is needed. That means, first it should check if the S3 location is already registered and when it is not, then it should register it.

*P.S. Don't attach files. Please, prefer add code snippets directly in the message body.*

**Comment by dlpzx** [comment]
I reviewed the [documentation](https://docs.aws.amazon.com/lake-formation/latest/dg/registration-role.html) to ensure this change would not affect sharing with data.all. 

From data.all we register the S3 location with the pivotRole. Lake Formation assumes the pivotRole to access the data in that S3 location. If the S3 location was previously registered with another IAM role the only difference is that under-the-hood Lake Formation uses a different role to access the data, but LF permissions work independently from the underlying IAM role that is assumed. 

In short, I confirm that we can use a pre-existing registration of the S3 location.
